cmd/di.go

```diff
@@ -468,23 +466,9 @@ func (di *Dependencies) bootstrapIdentityComponents(options node.Options) {
 
 func (di *Dependencies) bootstrapLocationComponents(options node.OptionsLocation, configDirectory string) (err error) {
 	di.IPResolver = ip.NewResolver(options.IPDetectorURL)
-
-	switch options.Type {
-	case "builtin":
-		di.LocationResolver, err = location.NewBuiltInResolver(di.IPResolver)
-	case "mmdb":
-		di.LocationResolver, err = location.NewExternalDBResolver(filepath.Join(configDirectory, options.Address), di.IPResolver)
-	case "oracle":
-		di.LocationResolver = location.NewOracleResolver(options.Address)
-	case "manual":
-		di.LocationResolver = location.NewStaticResolver(options.Country, options.City, options.NodeType, di.IPResolver)
-	default:
-		return fmt.Errorf("unknown location detector type: %s", options.Type)
-	}
-
+	di.LocationResolver, err = location.CreateLocationResolver(di.IPResolver, options.Country, options.City, options.Type, options.NodeType, options.Address, options.ExternalDb, configDirFlag)
```

Comment:
We are passing every single option separately, maybe it's time to pass just `OptionsLocation`?

Reply:
i'll ahve to ref the node package, which in turn add cyclic dependencies. It's either move the model somewhere, or define a new one here and use that instead. Neither seems to be a proper way of doing things :)